serviceaccount_test.go

```diff
@@ -0,0 +1,130 @@
+package wavefront
+
```

Comment:
Generally, unit tests are great, but I am not sure this type of unit test brings much value because without a wavefront server you have to mock everything.  All you can really test is that the request is getting marshaled into JSON correctly and that the mock response from the wavefront server gets unmarshaled correctly.  Verifying that the request marshaled correctly is tedious and can be hard to do especially if the go struct being marshaled contains embedded slices and maps, so usually our unit tests skip this.  So all that is left is checking that the mock response unmarshaled correctly.  Most bugs happen not because the go structs are being marshaled/unmarshaled incorrectly, but because wavefront server expects a different request than the API is giving or the API expects a different response than the wavefront server is giving, so when I write new APIs I find that I have to test them manually against a real wavefront server. 

I'd love to hear your thoughts on this and if you have any ideas about how to make testing these APIs better and more automated.  If the wavefront API server were written in Go, it may be possible to build a fake wavefront server for unit tests that resides in memory, but since the wavefront server is written in JAVA that becomes more difficult.

Reply:
I could have written that response myself: I absolutely agree with all of it. The tests in this PR were, again, imitations of the ones already there. I felt that they added little beyond proving the code compiled, but because the other clients had them, I thought these should too. I couldn't file a PR with *no* tests, and clearly someone deemed these "better than nothing" in the past. So here they are.

In my Ruby SDK all I do is assert the state of the outgoing API request, and I'm not sure what else a third-party developer can be expected to do. The only option, I guess, is to have acceptance tests that operate against a real endpoint and assert that what happens is correct. I've written these in the past and they're a lot of work, very slow to run, and hard to sandbox.

I can't really offer anything better than what's here, and I think it's marginally better than nothing.

Reply:
What you say sounds good to me.